### DISCUSSION TITLE
Thermostat stops before reaching target temperature

### DISCUSSION THREAD

**Discussion by celestianx** [jmcollin78/versatile_thermostat#657] [Q&A] [answered]
Hi, 

Since the beginning of winter, I have observed that when I set a target temperature, Versatile Thermostat will always stop 0.2-0.4 degrees below it. It wasn't behaving this last winter and not sure if this is a normal behaviour.
If this isn't normal, what can I do to fix it?
<img width="437" alt="image" src="https://github.com/user-attachments/assets/3bffd946-aa16-47e0-8144-e0352b1f95d1">
Thanks for your support

**Comment by jmcollin78** [comment]
Hello,

For this kind of issues I need:
- your VTherm attributes: (https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#custom-attributes)
- some regulation curves: (https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#even-better-with-plotly-to-tune-your-thermostat)

**Reply by celestianx** [reply]
Hi,

Thanks taking the "Séjour", the 
VThem Attribute :
```
hvac_modes: heat, off
min_temp: 7
max_temp: 25
target_temp_step: 0.1
preset_modes: none, frost, eco, comfort, boost
current_temperature: 18.9
temperature: 19
hvac_action: idle
preset_mode: none
is_on: true
hvac_mode: heat
type: null
is_controlled_by_central_mode: true
last_central_mode: Auto
frost_temp: 7
eco_temp: 17
boost_temp: 21
comfort_temp: 19
frost_away_temp: 0
eco_away_temp: 0
boost_away_temp: 0
comfort_away_temp: 0
power_temp: null
target_temperature_step: 0.1
ext_current_temperature: 4.5
ac_mode: false
current_power: null
current_power_max: null
saved_preset_mode: none
saved_target_temp: 19
saved_hvac_mode: heat
motion_sensor_entity_id: null
motion_state: null
power_sensor_entity_id: null
max_power_sensor_entity_id: null
overpowering_state: null
presence_sensor_entity_id: null
presence_state: null
window_state: off
window_auto_state: off
window_bypass_state: false
window_sensor_entity_id: null
window_delay_sec: null
window_auto_enabled: false
window_auto_open_threshold: null
window_auto_close_threshold: null
window_auto_max_duration: null
window_action: window_turn_off
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: 2024-11-22T19:39:24.392654+01:00
last_ext_temperature_datetime: 2024-11-22T18:17:18.722404+01:00
security_state: false
minimal_activation_delay_sec: 10
device_power: 1
mean_cycle_power: 0.23
total_energy: 76.5
last_update_datetime: 2024-11-22T19:52:01.137505+01:00
timezone: Europe/Paris
temperature_unit: °C
is_device_active: false
ema_temp: 18.77
is_used_by_central_boiler: false
temperature_slope: 0.46
hvac_off_reason: null
is_over_switch: true
is_inversed: false
keep_alive_sec: 0
underlying_entities: switch.radiateur_sejour_pw
on_percent: 0.23
power_percent: 23
on_time_sec: 70
off_time_sec: 229
cycle_min: 5
function: tpi
tpi_coef_int: 0.6
tpi_coef_ext: 0.01
friendly_name: Séjour
supported_features: 401
```

<img width="707" alt="image" src="https://github.com/user-attachments/assets/9b232f8a-731f-408a-927f-a393031e6e73">

**Comment by jmcollin78** [comment, marked as answer]
Hello @celestianx,

You should read this: https://github.com/jmcollin78/versatile_thermostat/blob/main/README-fr.md#type-over_switch-ou-over_valve
or this: https://www.hacf.fr/optimisation-versatile-thermostat/#optimiser-vos-vtherm

In synthesis: all is working fine but you just should tune your TPI coefficients. In your case, increase the coef_ext (x 2)

**Comment by celestianx** [comment]
Thanks for your quick feedback jm! Will look it.
